lib/resty/dns/resolver.lua

```diff
@@ -720,6 +750,32 @@ local function _tcp_query(self, query, id)
     return answers
 end
 
+local function _filter_sections(answers, opts)
+    local result = {}
+    local authority_section = false
+    local additional_section = false
+
+    if answers.errcode then
+      return answers
+    end
+
+    if opts then
+        authority_section = opts.authority_section
+        additional_section = opts.additional_section
+    end
+
+    for i, ans in ipairs(answers) do
+        if ans.section == SECTION_AN then
+           table.insert(result, ans)
+        elseif ans.section == SECTION_NS and authority_section then
+          table.insert(result, ans)
+        elseif ans.section == SECTION_AR and additional_section then
+          table.insert(result, ans)
+        end
+    end
```

Comment:
@kiddkai ,good job!

- the size of answers is explicitly, it's better to pre-allocate the `result` using `new_tab`.
- `table.insert` is expensive, it's better to replace using index.